File: examples/kotlin/features/workflow-scenario/kotlin/cott_impl/curriculum/workflow_scenario/apply_search.kt
package cott_impl.curriculum.workflow_scenario

internal fun apply_search(snapshot: curriculum.workflow_scenario.SearchSnapshot, candidate: curriculum.workflow_scenario.SearchResult): curriculum.workflow_scenario.SearchSnapshot {
    if (candidate.request_id != snapshot.request_id) return snapshot

    return curriculum.workflow_scenario.SearchSnapshot(
        request_id = snapshot.request_id,
        applied_request_id = candidate.request_id,
        query = snapshot.query,
        result = candidate.result,
        status = curriculum.workflow_scenario.SearchStatus.Ready
    )
}

File: examples/features/workflow-scenario/python/_cott_impl/curriculum/workflow_scenario/apply_search.py
from curriculum.workflow_scenario_types import SearchResult, SearchSnapshot, SearchStatus_Ready


def apply_search(snapshot: SearchSnapshot, candidate: SearchResult) -> SearchSnapshot:
    if candidate.request_id != snapshot.request_id:
        return snapshot
    return SearchSnapshot(request_id=snapshot.request_id, applied_request_id=candidate.request_id, query=snapshot.query, result=candidate.result, status=SearchStatus_Ready())
